packages/itwin/ec3-widget/src/components/EC3/EC3Config.ts

```diff
@@ -34,27 +33,39 @@ export interface EC3ConfigProps {
    * The base path for the Carbon Calculation API endpoints. If not specified, it defaults to {@link CARBON_CALCULATION_BASE_PATH}.
    */
   carbonCalculationBasePath?: string;
+}
 
+export type EC3ConfigPropsWithRedirectUri = EC3ConfigCommonProps & {
   /**
-   * A callback function that returns an access token for authenticating API requests.
-   * If not specified, it defaults to the authorizationClient of the {@link IModelApp}.
+   * The OAuth redirect URI used to authenticate with the EC3 API.
    */
-  getAccessToken?: GetAccessTokenFn;
-}
+  redirectUri: string;
+};
+
+export type EC3ConfigPropsWithGetEC3AccessToken = EC3ConfigCommonProps & {
+  /**
+   * A callback function that returns an access token for the EC3 API.
+   */
+  getEC3AccessToken: GetAccessTokenFn;
+};
+
+export type EC3ConfigProps = EC3ConfigPropsWithRedirectUri | EC3ConfigPropsWithGetEC3AccessToken;
 
 export class EC3Config {
```

Comment:
It would be best if this class would expose a method to `getAccessToken` regardless of which property (redirectUri or getEC3AccessToken) is provided. In the case of `getEC3AccessToken` it would be simple forward. In the case of redirectUri it would need to kick off the sign in process and wait for it to be completed.

Reply:
Sorry, I do not follow. `getAccessToken` is never undefined within the class and is a common prop between the two types. However, I moved the sign in process into the client with the same fallback logic.

Reply:
Sorry, by the `getAccessToken` I meant the method to get the access token for EC3, not for iTwinPlatform. As I understand the current `getAccessToken` method is for retrieving access token for iTwinPlatform. There is nothing that encapsulates the differences between `EC3ConfigPropsWithRedirectUri` and `EC3ConfigPropsWithGetEC3AccessToken`.

Reply:
Oh, I see you removed the public redirectUri and made the `getEC3AccessToken` handle it. That's what I wanted.